flink-connector-examples/src/main/java/io/pravega/example/flink/watermark/reader/PravegaWatermarkIngestion.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2018 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ */
+package io.pravega.example.flink.watermark.reader;
+
+import io.pravega.client.EventStreamClientFactory;
+import io.pravega.client.admin.StreamManager;
+import io.pravega.client.stream.EventStreamWriter;
+import io.pravega.client.stream.EventWriterConfig;
+import io.pravega.client.stream.ScalingPolicy;
+import io.pravega.client.stream.StreamConfiguration;
+import io.pravega.client.stream.impl.JavaSerializer;
+import io.pravega.connectors.flink.PravegaConfig;
+import io.pravega.example.flink.watermark.Constants;
+import io.pravega.example.flink.watermark.SensorData;
+import org.apache.flink.api.java.utils.ParameterTool;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URI;
+
+/*
+ * Parameters
+ *     -controller tcp://<controller_host>:<port>, e.g., tcp://localhost:9090
+ *
+ * Functionality
+ *     - Ingest simulated sine-wave sensor data with event timestamp into a Pravega stream(watermark-examples/input).
+ */
+public class PravegaWatermarkIngestion {
+
+    private static final Logger LOG = LoggerFactory.getLogger(PravegaWatermarkIngestion.class);
+
+    private static final double EVENT_VALUE_INCREMENT = 0.01; // Should be < 1
+    private static final int WRITER_SLEEP_MS = 100;
+
+    public static void main(String[] args) throws InterruptedException {
+        ParameterTool params = ParameterTool.fromArgs(args);
+        // The writer will contact with the Pravega controller to get information about segments.
+        URI pravegaControllerURI = URI.create(params.get(Constants.CONTROLLER_ADDRESS_PARAM, Constants.CONTROLLER_ADDRESS));
+        final int numEvents = Constants.EVENTS_NUMBER;
+
+        // StreamManager helps us to easily manage streams and copes.
+        StreamManager streamManager = StreamManager.create(pravegaControllerURI);
+
+        // A scope is a namespace that will be used to group streams (e.g., like dirs and files).
+        streamManager.createScope(Constants.DEFAULT_SCOPE);
```

Comment:
Don't create scope; it is assumed to exist.   `createScope` fails if the scope already exists right?

Reply:
No. The failure only happens if we have security enabled, this won't be the case for open-source community. It will return false not exception when the scope exists. See:
```
In io.pravega.client.stream.impl.ControllerImpl
            case SCOPE_EXISTS:
                log.warn(requestId, "Scope already exists: {}", new Object[]{scopeName});
                return false;
            case SUCCESS:
                log.info(requestId, "Scope created successfully: {}", new Object[]{scopeName});
                return true;
```